src/docs/schema/triggers.py

```diff
@@ -0,0 +1,117 @@
+from typing import List, Dict, Any
+from pydantic import BaseModel, validator, root_validator, Extra
+
+
+class ConfigMapFile(BaseModel):
+    """Schema of the microk8s[configmap][files] section."""
+
+    key: str
+    name: str
+    link: str
+    class Config:
+        extra = Extra.forbid
+
+
+class Microk8sConfigMap(BaseModel):
+    """Schema of the microk8s[configmap] section."""
+
+    name: str = None
+    files: List[ConfigMapFile]
+    class Config:
+        extra = Extra.forbid
+
+
+class Microk8sDeploy(BaseModel):
+    """Schema of the microk8s[deploy] section."""
+
+    link: str
+    access: str
+
+    class Config:
+        extra = Extra.forbid
+
+
+class Microk8sInfo(BaseModel):
+    """Schema of the microk8s section."""
+
+    configmap: Microk8sConfigMap = None
+    deploy: Microk8sDeploy
+
+    class Config:
+        extra = Extra.forbid
+
+
+^class DockerRunParameters(BaseModel):
+    """Schema of the docker section."""
+
+    parameters: List[str]
+    access: str = None
+
+    class Config:
+        extra = Extra.forbid
+
+
+class Parameter(BaseModel):
+    """Schema of the parameters section."""
+
+    type: str
+    value: str
+    description: str = None
+
+    class Config:
+        extra = Extra.forbid
+
+
+class DebugInfo(BaseModel):
+    """Schema of the debug section."""
+
+    text: str
+
+    class Config:
+        extra = Extra.forbid
+
+
+class ContainerImageInfo(BaseModel):
+    """Schema of the documentation.yaml file, and also validation of the schema"""
+
+    version: int
+    application: str
+    is_chiselled: bool = False
+    description: str
+    docker: DockerRunParameters = None
+    parameters: List[Parameter] = None
+    debug: DebugInfo = None
+    microk8s: Microk8sInfo = None
+    class Config:
+        extra = Extra.forbid
+
+    @validator("application")
+    def validate_application(cls, value):
```

Comment:
instead of using validators, just use "Field" above, setting the min_length for each. See https://docs.pydantic.dev/latest/usage/types/

Reply:
same for the other below